File: python/template.py
from python.constants import Constants


class Template:
    @staticmethod
    def male(championships: list):
        obj = {}
        for field in Constants.male_fields:
            obj[field] = 0
        for championship in championships:
            obj[championship] = 0
        return obj

    @staticmethod
    def female(championships: list):
        obj = {}
        for field in Constants.female_fields:
            obj[field] = 0
        for championship in championships:
            obj[championship] = 0
        return obj

File: javascript/template.js
const constants = require('./constants');
const FileHandler = require('./file');

const championships = FileHandler.readJson('championships');

class Template {
  static male() {
    const obj = {};
    constants.maleFields.forEach(field => obj[field] = 0);
    championships.primary.forEach(c => obj[c] = 0);
    championships.secondary.forEach(c => obj[c] = 0);
    championships.tertiary.forEach(c => obj[c] = 0);
    championships.tag_team.forEach(c => obj[c] = 0);
    return obj;
  }

  static female() {
    const obj = {};
    constants.femaleFields.forEach(field => obj[field] = 0);
    championships.women.forEach(c => obj[c] = 0);
    championships.women_tag_team.forEach(c => obj[c] = 0);
    return obj;
  }
}

module.exports = Template;
